### PR TITLE
Add usePagination hook

### PR DESCRIPTION
**Pull request by Elbarae1921**
Add react hook to abstract some of the jsx and directly return the pagination component.

### CODE REVIEW

File: stories/Paginate.stories.tsx [hunk 1]
```diff
@@ -24,14 +24,18 @@ const Template: Story<Props> = (args) => {
...
-    <Paginate
-      current={current}
-      handlePagination={handlePagination}
-      range={args.range}
-      total={args.total}
-      components={args.components}
-    />
+    <>
+      { pages }
+    </>
```
Comment (smakosh) [inline stories/Paginate.stories.tsx:67]: Let's better add another template rather override the default one, so developers can check the docs whether to use this hook or the Paginate component

**Review by smakosh — CHANGES_REQUESTED** [review]
Thanks for the contribution @Elbarae1921 🙌 , I left a comment, happy to elaborate if not understood!

File: stories/Paginate.stories.tsx (hunk 1 above)
Comment (Elbarae1921) [reply stories/Paginate.stories.tsx:67]: On it 👍.

Comment (Elbarae1921) [reply stories/Paginate.stories.tsx:38]: On it 👍.

**Comment by smakosh** [comment]
Can you resolve the conflicts @Elbarae1921 ?

**Comment by smakosh** [comment]
Could you add an empty commit following semantic commit convention? `git commit --allow-empty -m "feat(hook): implement a usePagination hook"`?

**Comment by Elbarae1921** [comment]
sure thing

**Review by smakosh — APPROVED** [review]
LGTM

**Comment by smakosh** [comment]
@all-contributors please add @Elbarae1921 for code and doc

**Pull request merged**